python/pyiceberg/expressions/__init__.py

```diff
@@ -111,7 +140,16 @@ class Reference(UnboundTerm[T]):
 
     name: str
 
-    def bind(self, schema: Schema, case_sensitive: bool = True) -> BoundReference[T]:
+    def __init__(self, name: str):
+        self.name = name
+
+    def __repr__(self):
+        return f"Reference(name={repr(self.name)})"
+
+    def __eq__(self, other):
+        return self.name == other.name if isinstance(other, Reference) else False
+
+    def bind(self, schema: Schema, case_sensitive: bool = True) -> BoundReference:
```

Comment:
This should return a typed bound reference, right?

Reply:
Added! 👍🏻